### PR TITLE
Issue #262: Fix type hints and functions signature

### LINKED ISSUE #262: Fix type hints and missing functions parameters.
We should complete the type hinting almost everywhere.

Also, the hook `hook_atomium_theme_hook()` is supposed to get 4 parameters and it's not used at all.

### PR DESCRIPTION
**Pull request by drupol**
Fix #262

### CODE REVIEW

File: atomium/includes/registry.inc [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * @see atomium_theme()
  * @see hook_theme()
  */
-function _atomium_theme(&$existing, $type, $theme, $path) {
+function _atomium_theme(array &$existing, $type, $theme, $path) {
```
Comment (aritomelo) [inline atomium/includes/registry.inc]: Accordingly, to what we already discussed, it needs to change the parameters, it should not be passed by reference as it is not to be overridden. please check  [here](https://github.com/drupal/drupal/blob/7.x/includes/theme.inc#L534)

**Review by aritomelo — CHANGES_REQUESTED** [review]
Check the comment, the rest is Great.

File: atomium/includes/registry.inc (hunk 1 above)
Comment (drupol) [reply atomium/includes/registry.inc]: Thanks, fixed!

**Review by aritomelo — APPROVED** [review]
Ready to be merged.

**Pull request merged**